src/cmd_pay.rs

```diff
@@ -1,21 +1,38 @@
-use crate::{result::Result, wallet::Wallet};
-use helium_proto::txns::blockchain::*;
-use prost::Message;
+use crate::{
+    keypair::PubKeyBin,
+    result::Result,
+    traits::{Sign, B58},
+    wallet::Wallet,
+};
+use helium_api::Client;
+use helium_proto::txn::{TxnPaymentV1, Wrapper};
+use prettytable::Table;
 
 pub fn cmd_pay(wallet: &Wallet, password: &str, payee: String, amount: u64) -> Result {
+    let client = Client::new();
+
     let keypair = wallet.to_keypair(password.as_bytes())?;
+    let account = client.get_account(&keypair.public.to_b58()?)?;
 
-    let payment = TxnPaymentV1 {
+    let mut txn = TxnPaymentV1 {
         amount,
         fee: 0,
-        payee: payee.as_bytes().to_vec(),
-        payer: Vec::new(),
-        nonce: 1,
+        payee: PubKeyBin::from_b58(payee)?.to_vec(),
+        payer: keypair.pubkey_bin().to_vec(),
+        nonce: account.nonce + 1,
```

Comment:
Why the `+ 1`?

Reply:
Rules of the API.. when you submit a new transaction you give it your account nonce + 1.. which maintains ordering..